lightningd/pay.c

```diff
@@ -1190,6 +1191,7 @@ static struct command_result *json_sendonion(struct command *cmd,
 		   p_opt("shared_secrets", param_secrets_array, &path_secrets),
 		   p_opt_def("partid", param_u64, &partid, 0),
 		   p_opt("bolt11", param_string, &b11str),
+		   p_opt("destination", param_node_id, &destination),
```

Comment:
Need documentation in `doc/lightning-sendonoin.7.md` as well,

Reply:
Also, you should not insert arguments, only append them.  This will conflict if the user uses an array as parameter instead of a JSON object.

Reply:
@ZmnSCPxj, mh! can clarify this case? I'm missing something here

Reply:
You should never insert a new parameter between existing ones (the exception is for dev-only parameters gated by `DEVELOPER`, which should always be at the end of the parameter list, and new non-dev parameters are inserted before that).  You can only append a new parameter.  So you should have put `destination` *after* `msatoshi`, not before `msatoshi`.  Latest code does this already.

Reply:
Oh! Thanks to clarify it. The next time I will pay more focus about it.